src/documents/DataSourceAPI.md

```diff
@@ -0,0 +1,135 @@
+transformation may be added before or after the encoding phase. For example providing
+[modelTransformations](ModelTransformationAPI.md) before/after serializing/deserializing JSON.
+
+### fluid.dataSource.parseJSON(string)
+
+Deserializes a JSON string into a proper JavaScript object and returns a promise for the result.
+
+* `string: {String}` The JSON string to be deserialized
+* Returns: `{Promise}` If there is an error, the promise will be rejected and contain the error message. Otherwise the
+promise will be resolved with the JavaScript object or `undefined` if the `string` value is falsey.
+
+### fluid.dataSource.stringifyJSON(obj)
+
+Serializes a JavaScript object into a JSON string.
+
+* `obj: {Object}` The JavaScript object to be serialized
+* Returns: `{String}` If the object is undefined, an empty string is returned.
+
+## Core
+
+### fluid.dataSource.registerStandardPromiseHandlers(that, promise, options)
+
+Registers the default promise handlers for a dataSource operation:
+
+1. If the user has supplied a function in place of method `options`, register this function as a success handler
+2. If the user has supplied an `onError` handler in method `options`, this is registered, otherwise we register the
+firer of the dataSource's own `onError` method.
+
+* `that {Object}` The concrete dataSource component
+* `promise {Promise}` The promise to bind the resolution and reject handlers too
+* `options {Object|Function}` (optional) Either a function to be executed on resolution, or an object containing an
+`onError` ({onError: function}) handler executed on rejection.
+
+### fluid.dataSource.defaultiseOptions(componentOptions, options, directModel, isSet)
+
+Compiles the options to be passed along to the each listener in a transform event sequence. The following properties
+will be added to the `options` passed in: `directModel`, `operation` ("set" or "get"), and `notFoundIsEmpty`.
+
+* `componentOptions {Object}` The components options.
+* `options {Object}` The base set of options to be passed along to the each listener.
+* `directModel {Object}` The direct model expressing the "coordinates" of the model
+* `isSet {Boolean}` A flag to indicate if it is for a "get" or "set" operation.
+
+### fluid.dataSource.get(that, directModel, options)
+
+Operate the core "transforming promise workflow" of a dataSource's `get` method. The initial listener provides the
+initial payload; which then proceeds through the transform chain to arrive at the final payload. A promise is returned
+with the final resolved payload.
+
+* `that {Component}` The dataSource itself
+* `directModel {Object}` The direct model expressing the "coordinates" of the model to be fetched
+* `options {Object}` A structure of options configuring the action of this get request - many of these will be specific
+to the particular concrete DataSource
+
+### fluid.dataSource.set(that, directModel, model, options)
```

Comment:
As above for fluid.dataSource.get - this in itself isn't an API function although it does basically form the contract of <dataSource>.set